tests/test_views.py

```diff
@@ -116,5 +122,49 @@ def test_response_404_when_none_specified(self):
             content = json.loads(response.content.decode('utf-8'))
             self.assertCountEqual({'message': 'No checks found', 'error': 404}, content)
 
+    @override_settings(WATCHMAN_LOGIN=False)
+    def test_login_not_required(self):
+        # Reload settings - and all dependent modules - from scratch
+        reload(sys.modules['watchman.settings'])
+        reload(sys.modules['watchman.decorators'])
+        reload(sys.modules['watchman.views'])
```

Comment:
Yes, I think we could refactor this into a function, but I don't think that's strictly necessary because the three tests are the only times these modules will be reloaded. Let me know what you want me to do with this.

Reloading the modules is necessary to pick up the overridden settings, because simply overriding the settings doesn't re-run the

    WATCHMAN_LOGIN = getattr(settings, 'WATCHMAN_LOGIN', False)

logic in `settings.py`, or the

    if settings.WATCHMAN_LOGIN:
        ....

logic in `decorators.py`, or the

    from watchman.decorators import token_required, login_required

logic in `views.py`, but re-importing all of those modules in that specific order does.

The [`@override_settings` decorator](https://docs.djangoproject.com/en/1.4/topics/testing/#django.test.utils.override_settings) automatically ensures that the settings do not propagate outside of the decorated test/s (it can also decorate classes).